CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -0,0 +1,535 @@
+                .ReturnsAsync(new List<long>() { 2 });
+
+            dashbordrepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAsync(
+                (long)studentResultRequestWithOutStudent.CourseId, 
+                studentResultRequestWithOutStudent.StartDate,
+                studentResultRequestWithOutStudent.FinishDate))
+                .ReturnsAsync(default(List<long>));
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>() { 2 }))
+                .ReturnsAsync(new List<long>() { 6, 7, 8, 9, 10 });
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(default(List<long>)))
+                .ReturnsAsync(default(List<long>));
+            
+            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(
+               new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+                .ReturnsAsync(new List<AverageStudentMarkDto>() {
+                    new AverageStudentMarkDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageMark = 5.1
+                    }
+                });
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(default(List<long>), default(List<long>)))
+                .ReturnsAsync(new List<AverageStudentMarkDto>());
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(
+                 new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+                .ReturnsAsync(new List<AverageStudentVisitsDto>()
+                {
+                    new AverageStudentVisitsDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageVisitsPercentage = 15
+                    }
+                });
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(default(List<long>), default(List<long>)))
+                .ReturnsAsync(new List<AverageStudentVisitsDto>());
+
+            _unitOfWorkMock.Setup(x => x.DashboardRepository).Returns(dashbordrepositoryMock.Object);
+
+            var dashbordservice = new DashboardService(_unitOfWorkMock.Object);
+
+            //Act
+            var resultWithData = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithData);
+            var resultWithWrongData = await dashbordservice.GetStudentsResultAsync(studentResultWrongRequest);
+            var resultWithoutStudent = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithOutStudent);
+
+            //Assert
+            Assert.NotNull(resultWithData.Data);
```

Comment:
I think better to use this check. As in most cases it will not return null. Only in case of method error
```
            Assert.NotEmpty(resultWithData.Data.AverageStudentsMarks);
            Assert.NotEmpty(resultWithData.Data.AverageStudentVisits);
```